projects/driver_controls/src/ads1015.c

```diff
@@ -0,0 +1,169 @@
+#include "ads1015.h"
+
+#include <stdbool.h>
+#include <stdio.h>
+
+#include "i2c.h"
+#include "gpio_it.h"
+#include "ads1015_bitmasks.h"
+
+// Union definition to help process 16-bit registers
+// Due to endian-ness, index 0 refers to the most significant byte
+typedef union ADS1015Register {
+  uint16_t raw;
+  uint8_t data[2];
+} ADS1015Register;
+
+typedef struct ADS1015Interrupt {
+  ADS1015Callback callback;
+  void *context;
+  int16_t reading;
```

Comment:
Why is reading stored as a signed integer? Also, please make it more clear whether this reading is raw or converted.

It would also be nice to add a comment to the read functions that explicitly states whether it begins another conversion and reads that data or uses stored data.